src/frontend/src/catalog/root_catalog.rs

```diff
@@ -651,7 +652,24 @@ impl Catalog {
             .ok_or_else(|| CatalogError::NotFound("table", table_name.to_string()))
     }
 
-    pub fn get_table_by_id(&self, table_id: &TableId) -> CatalogResult<&Arc<TableCatalog>> {
+    /// Used to get `TableCatalog` for Materialized Views, Tables and Indexes.
+    /// Retrieves only created tables.
+    pub fn get_created_table_by_name<'a>(
```

Comment:
`SchemaCatalog` has a `iter_table` interface. I think we should change it to `iter_created_table` as well.

Reply:
Done